tests/unit/notes/test-class-wc-payments-notes-set-up-stripelink.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Class WC_Payments_Notes_Set_Up_StripeLink_Test
+ *
+ * @package WooCommerce\Payments\Tests
+ */
+
+use WCPay\Session_Rate_Limiter;
+
+/**
+ * Class WC_Payments_Notes_Set_Up_StripeLink tests.
+ */
+class WC_Payments_Notes_Set_Up_StripeLink_Test extends WCPAY_UnitTestCase {
+	/**
+	 * System under test.
+	 *
+	 * @var WC_Payment_Gateway_WCPay
+	 */
+	private $mock_wcpay_gateway;
+
+	/**
+	 * Mock WC_Payments_Customer_Service.
+	 *
+	 * @var WC_Payments_Customer_Service|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_customer_service;
+
+	/**
+	 * Mock WC_Payments_Token_Service.
+	 *
+	 * @var WC_Payments_Token_Service|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_token_service;
+
+	/**
+	 * Mock WC_Payments_API_Client.
+	 *
+	 * @var WC_Payments_API_Client|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_api_client;
+
+	/**
+	 * Mock WC_Payments_Action_Scheduler_Service.
+	 *
+	 * @var WC_Payments_Action_Scheduler_Service|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_action_scheduler_service;
+
+	/**
+	 * Mock Session_Rate_Limiter.
+	 *
+	 * @var Session_Rate_Limiter|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_rate_limiter;
+
+	/**
+	 * WC_Payments_Order_Service.
+	 *
+	 * @var WC_Payments_Order_Service
+	 */
+	private $order_service;
+
+	/**
+	 * Mock WC_Payments_Account.
+	 *
+	 * @var WC_Payments_Account|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_wcpay_account;
+
+	public function set_up() {
+		parent::set_up();
+
+		require_once WCPAY_ABSPATH . 'includes/notes/class-wc-payments-notes-set-up-stripelink.php';
+
+		$this->mock_api_client = $this->getMockBuilder( 'WC_Payments_API_Client' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->mock_wcpay_account = $this->createMock( WC_Payments_Account::class );
+
+		$this->mock_customer_service = $this->getMockBuilder( 'WC_Payments_Customer_Service' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->mock_token_service = $this->getMockBuilder( 'WC_Payments_Token_Service' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->mock_action_scheduler_service = $this->getMockBuilder( 'WC_Payments_Action_Scheduler_Service' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$this->mock_rate_limiter = $this->createMock( Session_Rate_Limiter::class );
+
+		$this->order_service = new WC_Payments_Order_Service( $this->mock_api_client );
+
+		$this->mock_wcpay_gateway = $this->getMockBuilder( '\WC_Payment_Gateway_WCPay' )
```

Comment:
Looking into the test case, we set some expectations on `mock_wcpay_gateway` only. Could we omit to introduce unneeded properties and relocate the mocks construction right into constructor arguments, perhaps?

Reply:
Makes sense. I have disabled the constructor and used a single mock.